src/mpyl/build.py

```diff
@@ -125,22 +119,20 @@ def run_mpyl(
 
 
 def find_build_set(
-    repo: Repository, changes_in_branch: list[Revision], build_all: bool
+    repo: Repository,
+    changes_in_branch: list[Revision],
+    stages: list[Stage],
+    build_all: bool,
 ) -> dict[Stage, set[Project]]:
     project_paths = repo.find_projects()
     all_projects = set(
         map(lambda p: load_project(Path(""), Path(p), False), project_paths)
     )
 
     if build_all:
-        return {
```

Comment:
I think we can already decide on which approach we want to use between this and https://github.com/Vandebron/mpyl/pull/281/files#diff-5f8bc6ee933a7bacafed256143948c5c0b696a67bb5cddab95f2e0f3578f9dbaR138-R153 . There is no merge conflict except in the picked approach. Here the stages get passed and in my pr they are still hardcoded from a function, but for the rest the only difference is the way in which we coded it.

Maybe @koeves can vote on which way he likes most 😉

Reply:
I thought we changed that to `[s for s in Stage]` or something like that in your PR. It should be possible to extract the stages in order from an Enum, that way it will be correct even if we add a new stage

Reply:
True, I don't know what happened with that change. I'll see if I can implement it

Reply:
The order of stages is defined in `run_properties.yml`